var/spack/repos/builtin/packages/rocm-tensile/package.py

```diff
@@ -33,29 +36,48 @@ class RocmTensile(CMakePackage):
         depends_on('rocm-smi@' + ver, type='build', when='@' + ver)
         depends_on('rocm-smi-lib@' + ver, type='build', when='@' + ver)
         depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
-
     # This is the default library format since 3.7.0
     depends_on('msgpack-c@3:', when='@3.7:')
     depends_on('boost@1.58.0', type=('build', 'link'))
+    depends_on('llvm-openmp', type='build')
 
     root_cmakelists_dir = 'Tensile/Source'
     # Status: https://github.com/ROCmSoftwarePlatform/Tensile/commit/a488f7dadba34f84b9658ba92ce9ec5a0615a087
     # Not yet landed in 3.7.0, nor 3.8.0.
-    patch('0001-fix-compile-error.patch')
+    patch('0001-fix-compile-error.patch', when='@3.7.0:3.8.0')
+
+    def patch(self):
+        if '@3.9.0:' in self.spec:
+            kwargs = {'ignore_absent': False, 'backup': False, 'string': False}
+            match = '/usr/lib/x86_64-linux-gnu/libomp.so'
+            substitute = '${OPENMP_LIBRARY}'
+            files = [
+                'Tensile/Source/CMakeLists.txt',
+                'Tensile/Source/client/CMakeLists.txt',
+                'HostLibraryTests/CMakeLists.txt'
+            ]
+            filter_file(match, substitute, *files, **kwargs)
 
     def setup_build_environment(self, env):
         env.set('CXX', self.spec['hip'].hipcc)
+        if '@3.9.0:' in self.spec:
+            lib_dir = self.spec['llvm-openmp'].prefix.lib
+            env.prepend_path('LIBRARY_PATH', lib_dir)
 
     def cmake_args(self):
-        archs = ",".join(self.spec.variants['amdgpu_target'].value)
-
+        arch = self.spec.variants['tensile_architecture'].value
+        boost_incl = self.spec['boost'].prefix.include
         args = [
             '-Damd_comgr_DIR={0}'.format(self.spec['comgr'].prefix),
             '-DTensile_COMPILER=hipcc',
-            '-DTensile_ARCHITECTURE={0}'.format(archs),
+            '-DTensile_ARCHITECTURE={0}'.format(arch),
             '-DTensile_LOGIC=asm_full',
             '-DTensile_CODE_OBJECT_VERSION=V3',
-            '-DBoost_USE_STATIC_LIBS=Off',
+            '-DBoost_USE_STATIC_LIBS=off',
+            '-DCMAKE_CXX_FLAGS:String=-I{0}'.format(boost_incl),
```

Comment:
Boost should really be found by cmake, and we should avoid adding compiler flags like this. What is the problem here you're trying to solve here?

Reply:
Iam running into below issue if i dont add  **'-DCMAKE_CXX_FLAGS:String=-I{0}'.format(boost_incl),**


```
cd /tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-build-xoe7npt/client && /tmp/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/hip-3.10.0-wysrhl5hb3y6345raqhdj3mg5jyaal7m/bin/hipcc -DTENSILE_DEFAULT_SERIALIZATION -DTENSILE_STATIC_ONLY -DTENSILE_USE_HIP -DTENSILE_YAML=1 -D__HIP_PLATFORM_HCC__=1 -D__HIP_ROCclr__=1 -I/tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/include -I/tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/lib/include -isystem /tmp/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/rocm-smi-lib-3.10.0-dn6j5fjh2imo6myb6qh54vmv4c4wdkmx/include -fopenmp -std=c++14 -O2 -g -DNDEBUG -mno-xnack -Xarch_gfx906 -msram-ecc -Xarch_gfx908 -msram-ecc -xhip --cuda-gpu-arch=gfx900 --cuda-gpu-arch=gfx906 --cuda-gpu-arch=gfx908 -o CMakeFiles/TensileClient.dir/source/TimingEvents.cpp.o -c /tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/source/TimingEvents.cpp
In file included from /tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/source/DataInitialization.cpp:27:
/tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/include/DataInitialization.hpp:29:10: fatal error: 'boost/program_options.hpp' file not found
#include <boost/program_options.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/source/CSVStackFile.cpp:27:
/tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/include/CSVStackFile.hpp:34:10: fatal error: 'boost/lexical_cast.hpp' file not found
#include <boost/lexical_cast.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/source/ResultFileReporter.cpp:27:
In file included from /tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/include/ResultFileReporter.hpp:29:
/tmp/root/spack-stage/spack-stage-rocm-tensile-3.10.0-xoe7npteuwocnb6hwo6r6xqslutqn4en/spack-src/Tensile/Source/client/include/CSVStackFile.hpp:34:10: fatal error: 'boost/lexical_cast.hpp' file not found
#include <boost/lexical_cast.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~
1 error generated when compiling for gfx900.
make[2]: *** [client/CMakeFiles/TensileClient.dir/build.make:98: client/CMakeFiles/TensileClient.dir/source/CSVStackFile.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 error generated when compiling for gfx900.
make[2]: *** [client/CMakeFiles/TensileClient.dir/build.make:137: client/CMakeFiles/TensileClient.dir/source/DataInitialization.cpp.o] Error 1
1 error generated when compiling for gfx900.

```

Reply:
Which version of `boost` are you using? Is the issue reproducible with other specs built with CMake against boost at the same version?

Reply:
This is not the way to fix that problem. Please figure out why https://github.com/ROCmSoftwarePlatform/Tensile/blob/257ff1f98d8cb0f6361b40f5313bdf9f94f4155b/Tensile/Source/client/CMakeLists.txt#L37 is not adding the include flags to those targets.

Reply:
@haampie Sure will check and create a separate PR for llvm-amdgpu recipe update and also add above patch. In the meantime, can we merge this in.

Reply:
I agree this should be looked at and requested at source. But this is a tagged release so even if get them to add the include directory there would require a patch for the current releases. Is my understanding correct? If yes, should we be able to merge this ?